### ISSUE TITLE
`input{some-filte.tex}` no longer works in `header-includes` for html output

### ISSUE THREAD

**Issue by lassepe** [jgm/pandoc#6319]
In **pandoc 2.7.3** I used be able to do something like this for `reveal.js` slides:

```pandoc
---
revealjs-url: https://revealjs.com
header-includes:
    - \input{macros.tex}
---

$v = \argmax{x}f(x)$

```

where `macros.tex` contains:

```
\newcommand{\argmax}[1]{\underset{#1}{\operatorname{arg\,max\,}}}
```

compiling with `pandoc --to revealjs --mathjax --standalone --output out.html  in.md` this correctly renders to:

![image](https://user-images.githubusercontent.com/10076790/80398257-9fd2b380-88b7-11ea-9fe0-aff2a6a66909.png)

However, in **pandoc 2.9.2.1** this no longer works. The macro is not expanded and results in this rendering:

![image](https://user-images.githubusercontent.com/10076790/80398199-892c5c80-88b7-11ea-911c-a5da35cf0942.png)

This seems to be an issue with `\input`. If I put the input of `macros.tex` directly in `header-includes`, i.e.:

```pandoc
header-includes:
    - \newcommand{\argmax}[1]{\underset{#1}{\operatorname{arg\,max\,}}}
```
Then things are still rendered correctly.

Is this intended behavior change from 2.7.3 to 2.9.2.1?

**Comment by alerque** [comment]
This might be an iteration of #5673 I raised a while back. Try passing a language tagged code block instead of a plain string:

```yaml
---
revealjs-url: https://revealjs.com
header-includes: |
    - |
    ```{=latex}
    \input{macros.tex}
    ```
---
```

**Comment by lassepe** [comment]
@alerque thank you for your quick reply.

Unfortunately, your suggested fix does not seem to help.

**Comment by jgm** [comment]
Try
```
---
revealjs-url: https://revealjs.com
header-includes: |
    - |
    ```{=html}
    \input{macros.tex}
    ```
---
```
Otherwise pandoc will not render it when rendering reveal.js, since it's marked as `latex`.

**Comment by lassepe** [comment]
Same result on master (82280cc62c9fdfe7ffe1b0d8443d1739b441e22b).

**Comment by jgm** [comment]
If you look at the generated HTML file, what do you see?  Is the line `\input{macros.tex}` there?
Pandoc would never have actually parsed the contents of macros.tex if this were included in header-includes.
Perhaps there was a relevant change in reveal.js, or mathjax?  (I assume you're including mathjax?)

**Comment by jgm** [comment]
Ah, well wait.  In fact pandoc does parse the contents of metadata, even when it's in a field like `header-includes`. So it's possible that in an earlier version it looked in the `\input` file and included the macro definition?

**Comment by jgm** [comment]
OK, pandoc 2.8 changelog has this item for the LaTeX reader:
```
    + Don't try to parse includes if `raw_tex` is set (#5673).
      When the `raw_tex` extension is set, we just carry through
      `\usepackage`, `\input`, etc. verbatim as raw LaTeX.
```
Maybe the change was ill considered, because it prevents the kind of thing you're doing, which used to work.  After all, `raw_tex` has to be set when reading markdown for latex to be processed at all (and it is set by default).  Unsetting it would not be an option here.

@alerque did we go too far in #5673?  After all, using the raw attribute was a viable solution there.
On the other hand, changing this now could break existing documents.

**Comment by lassepe** [comment]
> If you look at the generated HTML file, what do you see? Is the line `\input{macros.tex}` there?
> Pandoc would never have actually parsed the contents of macros.tex if this were included in header-includes.

Yes, you are right. If I mark the block as `{=html} the html document simply containts `\input{macros.tex}` in the header and does not actually inline anything. If I don't add the language tag (or if I add the language tag latex) then it'll not show up at all.
